### PR TITLE
Aged and Negative Quality test

### PR DESCRIPTION
**Pull request by ChusMoyano**

### CODE REVIEW

File: src/test/java/com/gildedrose/GildedRoseTest.java [hunk 1]
```diff
@@ -19,4 +19,31 @@ void fooTest() {
         assertEquals(40, app.items[0].quality);
     }
 
+
+   @Test
+   void agedBrieTest() {
```
Comment (SalloumG) [inline src/test/java/com/gildedrose/GildedRoseTest.java]: typically I like test names that are explicit of what is being tested. and alternative can be the @DisplayName tag provided by Junit (preferably both).

When reading the tests or when the tests fails in the logs, seeing something like (aged brie decreases in quality every day) is a better indicator of what the test is doing

File: src/test/java/com/gildedrose/GildedRoseTest.java [hunk 2]
```diff
@@ -19,4 +19,31 @@ void fooTest() {
...
+       Item[] items = new Item[] { new Item("Aged Brie", 5, 0) };
+       GildedRose gildedRose = new GildedRose(items);
+       gildedRose.updateQuality();
+
+       assertEquals("Aged Brie", gildedRose.items[0].name);
+       assertEquals(4, gildedRose.items[0].sellIn);
```
Comment (SalloumG) [inline src/test/java/com/gildedrose/GildedRoseTest.java]: it's nice to have a test scenario with one reason to fail. It's testing a unitary action. I would personally move the sellIn to another test. Or at least make the name more explicit.

But preferably one reason to fail/ test

File: src/test/java/com/gildedrose/GildedRoseTest.java [hunk 3]
```diff
@@ -19,4 +19,31 @@ void fooTest() {
...
+       assertEquals(1, gildedRose.items[0].quality);
+   }
+
+    @Test
+    void negativeQuality() {
```
Comment (SalloumG) [inline src/test/java/com/gildedrose/GildedRoseTest.java]: naming: I felt from the test name that we expect a negative quality.  adding something like @DispalyName(Quality cannot be negative) or renaming the test to quality_should_not_be_negative can be a better description

File: src/test/java/com/gildedrose/GildedRoseTest.java (hunk 1 above)
Comment (ChusMoyano) [reply src/test/java/com/gildedrose/GildedRoseTest.java]: Done

File: src/test/java/com/gildedrose/GildedRoseTest.java (hunk 2 above)
Comment (ChusMoyano) [reply src/test/java/com/gildedrose/GildedRoseTest.java]: Done

File: src/test/java/com/gildedrose/GildedRoseTest.java (hunk 3 above)
Comment (ChusMoyano) [reply src/test/java/com/gildedrose/GildedRoseTest.java]: Done

File: src/test/java/com/gildedrose/GildedRoseTest.java [hunk 4]
```diff
@@ -1,8 +1,10 @@
 package com.gildedrose;
 
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
 
 class GildedRoseTest {
```
Comment (SalloumG) [inline src/test/java/com/gildedrose/GildedRoseTest.java:9]: What's the purpose of the fooTest? we can remove them If they are no longer valid

File: src/test/java/com/gildedrose/GildedRoseTest.java [hunk 5]
```diff
@@ -19,4 +21,44 @@ void fooTest() {
...
+       Item[] items = new Item[] { new Item("Aged Brie", 5, 0) };
+       GildedRose gildedRose = new GildedRose(items);
+       gildedRose.updateQuality();
+
+       assertEquals("Aged Brie", gildedRose.items[0].name);
+       assertEquals(1, gildedRose.items[0].quality);
+   }
+
+    @Test
+    @DisplayName("Aged brie non increase quality everyday")
```
Comment (SalloumG) [inline src/test/java/com/gildedrose/GildedRoseTest.java]: what's the rule we are validating here?

File: src/test/java/com/gildedrose/GildedRoseTest.java (hunk 4 above)
Comment (ChusMoyano) [reply src/test/java/com/gildedrose/GildedRoseTest.java:9]: Removed

File: src/test/java/com/gildedrose/GildedRoseTest.java (hunk 5 above)
Comment (ChusMoyano) [reply src/test/java/com/gildedrose/GildedRoseTest.java]: Removed

**Pull request merged**